File: engine/src/main/java/de/hanno/hpengine/engine/component/Component.java
package de.hanno.hpengine.engine.component;

import de.hanno.hpengine.engine.entity.Entity;
import de.hanno.hpengine.engine.lifecycle.Updatable;

import java.io.Serializable;

public interface Component extends Updatable, Serializable {

	Entity getEntity();

	default void update(float seconds) {}

	String getIdentifier();

	default void destroy() { }
}

File: engine/src/main/java/de/hanno/hpengine/engine/component/Component.kt
package de.hanno.hpengine.engine.component

import de.hanno.hpengine.engine.entity.Entity
import de.hanno.hpengine.engine.lifecycle.Updatable
import de.hanno.hpengine.engine.scene.Scene
import java.io.Serializable

interface Component : Updatable, Serializable {

    val entity: Entity

    override suspend fun update(scene: Scene, deltaSeconds: Float) {}

    fun destroy() {}
}
